firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -508,8 +509,15 @@ class BrowserViewController: UIViewController,
         presentedViewController?.view.alpha = 0
     }
 
+    var canShowPrivacyView: Bool {
+        // Show privacy view if no view controller is presented
+        // or if the presented view is a PhotonActionSheet.
+        self.presentedViewController == nil || presentedViewController is PhotonActionSheet
+    }
+
     @objc
     func appDidBecomeActiveNotification() {
+        guard canShowPrivacyView else { return }
```

Comment:
I think this check needs to be more granular? I don't believe we should be skipping all of the code in this method, there is quite a lot that it does which isn't specific to the privacy view AFAIU. 🤔 

LMK if you believe that's incorrect, but this func also calls into `browserDidBecomeActive()` for example, which performs important state updates which we need to be sure are performed regardless of the UI/privacy screen or which VC is presented.

Reply:
Now that you mention I don't think it's necessary at all. This method restores to the initial state if the blur was added. Since this was also called before for tabs that are not private, I think it's fine to remove the check.